### PR TITLE
Change os dependent definitions to new syntax

### PR DESCRIPTION
**Pull request by andreasnoack**
and move them to the end of the source file.

I've also added a condition for the `chol` test to avoid the warning on newer versions.

### CODE REVIEW

File: src/Compat.jl [hunk 1]
```diff
@@ -1219,4 +1139,88 @@ else
     const KERNEL = Sys.KERNEL
 end
 
+import Base: srand, rand, rand!
+if VERSION < v"0.4.0-dev+1373" # PR #8854
+    function make_seed()
+        try
+            if is_unit()
```
Comment (tkelman) [inline src/Compat.jl]: typo

File: src/Compat.jl [hunk 2]
```diff
@@ -132,14 +132,18 @@ import Base: srand, rand, rand!
 if VERSION < v"0.4.0-dev+1373" # PR #8854
     function make_seed()
         try
-            @unix_only seed = open("/dev/urandom") do io
-                a = Array(UInt32, 4)
-                read!(io, a)
-                a
+            @static if is_unix()
```
Comment (yuyichao) [inline src/Compat.jl]: ....... note that I don't think `@static` is defined until much later ..... You can reverse the `if` and `function` if you don't want to move this.

File: src/Compat.jl [hunk 3]
```diff
@@ -130,19 +130,27 @@ end
...
+                seed = let a = zeros(UInt32, 2)
+                    Base.dSFMT.win32_SystemFunction036!(a)
+                    a
+                end
             end
             return seed
-        catch
+        end
+    catch
+        function make_seed()
```
Comment (yuyichao) [inline src/Compat.jl]: Errr..... the try catch should still be in the function....

**Comment by andreasnoack** [comment]
This one is ready. The AV error seems unrelated.

File: test/runtests.jl [hunk 4]
```diff
@@ -406,11 +406,13 @@ let s = "abcdef", u8 = "abcdef\uff", u16 = utf16(u8), u32 = utf32(u8),
     @test isvalid(UTF32String, u32)
 end
 
-# chol
-let A = rand(2,2)
-    B = A'*A
-    U = @compat chol(B, Val{:U})
-    @test_approx_eq U'*U B
+if VERSION < v"0.5.0-dev+907"
```
Comment (tkelman) [inline test/runtests.jl]: did this get deprecated?

File: src/Compat.jl [hunk 5]
```diff
@@ -1242,4 +1162,88 @@ else
...
+    function make_seed()
+        try
+            @static if is_unix()
+                seed = open("/dev/urandom") do io
+                    a = Array(UInt32, 4)
+                    read!(io, a)
+                    a
+                end
+            end
+            @static if is_windows()
```
Comment (tkelman) [inline src/Compat.jl]: better as an `else`

File: test/runtests.jl (hunk 4 above)
Comment (andreasnoack) [inline test/runtests.jl]: Yes. The new version only accepts the matrix and not an `uplo` argument.

Comment (tkelman) [inline test/runtests.jl]: is it just cholfact that lets you pick now, or only at the lapack level?

Comment (andreasnoack) [inline test/runtests.jl]: It is still allowed in `cholfact`. In contrast to the Cholesky factor, the factorization as such doesn't depend on the `uplo` argument so we still allow specifying `uplo` in `cholfact`.

File: src/Compat.jl (hunk 5 above)
Comment (andreasnoack) [inline src/Compat.jl]: Done!

**Pull request merged**